docs/dev-installation.rst

```diff
@@ -1,9 +1,161 @@
 
-Developer Installation
-======================
+Developer Installation Guide
+============================
 
-You can either install Girder natively on your machine or inside a virtual
-machine (VM) with Vagrant.
+If you wish to develop Girder or Girder plugins, you can either
+:ref:`install Girder natively <native_install>` on your machine or
+:ref:`inside a virtual machine (VM) <vagrant_install>` with Vagrant.
+
+.. _native_install:
+
+Native Installation
++++++++++++++++++++
+
+Before you install, see the :doc:`prerequisites` guide to make sure you
+have all required system packages installed.
+
+Creating a virtual environment
+------------------------------
+
+While not strictly required, it is recommended to install Girder within
+its own `virtual environment <http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_
+to isolate its dependencies from other python packages.  To generate a new
+virtual environment, first install/update the ``virtualenv`` and ``pip``
+packages ::
+
+   sudo pip install -U virtualenv pip
```

Comment:
Ideally, these instructions should be tested and extracted from a real shell script using `.. literalinclude::`. For example, see https://github.com/NeurodataWithoutBorders/pynwb/blob/b715e5523c3e2b2b1bd219c23a8b06cbe6cb9445/docs/source/example.rst#examples

Reply:
Cool feature, but out of scope for this PR.

Reply:
Fair enough